source/s3_client.c

```diff
@@ -1687,7 +1669,7 @@ static bool s_s3_client_should_update_meta_request(
     size_t num_known_vips = client->vtable->get_host_address_count(
         client->client_bootstrap->host_resolver, endpoint->host_name, AWS_GET_HOST_ADDRESS_COUNT_RECORD_TYPE_A);
     if (num_known_vips == 0 && (client->threaded_data.num_requests_being_prepared +
-                                client->threaded_data.request_queue_size) >= g_max_num_connections_per_vip) {
+                                client->threaded_data.request_queue_size) >= g_min_num_connections) {
```

Comment:
do we still want this?

Reply:
I don't know yes/no for certain. I wanted to limit the extent of these changes so I'm keeping the old behavior where we limit the number of requests in the prep stage when we don't know *any* IP addresses yet.

I also wondered if this was added in response to a specific issue, or just something that was added superstitiously. The check was added in this PR: [Multiple Bucket Support](https://github.com/awslabs/aws-c-s3/pull/136). There's a specific commit ([Preventing too many requests from being prepared](https://github.com/awslabs/aws-c-s3/pull/136/commits/19fd44f3be0e073b648d28ea0609031e25c04cf7)) where Ryan adds that adds this check. So it seems pretty deliberate.

So maybe this could be removed? Or the numbers tweaked? We'll address that another day